include/envoy/network/address.h

```diff
@@ -102,23 +102,34 @@ class Ip {
 };
 
 /**
- * Interface for a generic Pipe address
+ * Interface for a generic Pipe address.
  */
 class Pipe {
 public:
   virtual ~Pipe() = default;
   /**
-   * @return abstract namespace flag
+   * @return abstract namespace flag.
    */
   virtual bool abstractNamespace() const PURE;
 
   /**
-   * @return pipe mode
+   * @return pipe mode.
    */
   virtual mode_t mode() const PURE;
 };
 
-enum class Type { Ip, Pipe };
+class EnvoyInternalAddress {
+public:
+  virtual ~EnvoyInternalAddress() = default;
+
+  /**
+   * @return The unique id of the internal address. If the address represents the destination
```

Comment:
I think that  ListenerManagerImpl::hasListenerWithAddress should handle the case of verifying uniqueness, but can you add a listener unit test to regression test this (i.e. try to have two listeners with the same internal address?)  With pipes and actual addresses we can rely on things like bind() failing with duplicated addresses so I think it's worth checking that we never lose the equality test for internal listeners.

Reply:
I add the guard that reject the listener config if internal address is used in ListenerManagerImpl,and tested The rejection is more than checking the uniqueness.

I can add a TODO and adding in the immediately in the next WIP PR